lightning/src/chain/channelmonitor.rs

```diff
@@ -3353,20 +3416,24 @@ impl<Signer: WriteableEcdsaChannelSigner> ChannelMonitorImpl<Signer> {
 						matured_htlcs.push(source.clone());
 					}
 
-					log_debug!(logger, "HTLC {} failure update in {} has got enough confirmations to be passed upstream",
-						log_bytes!(payment_hash.0), entry.txid);
-					self.pending_monitor_events.push(MonitorEvent::HTLCEvent(HTLCUpdate {
-						payment_hash,
-						payment_preimage: None,
-						source: source.clone(),
-						htlc_value_satoshis,
-					}));
-					self.htlcs_resolved_on_chain.push(IrrevocablyResolvedHTLC {
-						commitment_tx_output_idx,
-						resolving_txid: Some(entry.txid),
-						resolving_tx: entry.transaction,
-						payment_preimage: None,
-					});
+					let duplicate_event = self.pending_monitor_events.iter().any(
+						|update| if let &MonitorEvent::HTLCEvent(ref upd) = update { upd.source == *source } else { false });
```

Comment:
Does this check for duplicate events belong in this commit?

Reply:
Before this commit we never had to check for duplicates since this was the only place we pushed the event and we wouldn't repeat it AFAICT, but this commit introduced a new place where we're pushing this event so that's why I included it in the same one. I could separate it out into it's own commit? I don't feel strongly either way so let me know

Reply:
Oh sorry, missed that. Okay definitely belongs here.